File: Level1/키패드 누르기/키패드 누르기.java
import java.util.*;
class Solution {
    public String solution(int[] numbers, String hand) {
        String answer = "";
        
        int[][] ph_nums = {{1,3},{0,0},{1,0},{2,0},{0,1},{1,1},{2,1},{0,2},{1,2},{2,2}};
        int[] curr_left = {0,3};
        int[] curr_right = {2,3};
        
        for(int i=0; i<numbers.length; i++){
            if(numbers[i] == 1 || numbers[i] == 4 || numbers[i] == 7){
                curr_left = ph_nums[numbers[i]];
                answer += "L";
            }else if(numbers[i] == 3 || numbers[i] == 6 || numbers[i] == 9){
                curr_right = ph_nums[numbers[i]];
                answer += "R";
            }else{
                int ch_left = Math.abs(curr_left[0]-ph_nums[numbers[i]][0])+Math.abs(curr_left[1]-ph_nums[numbers[i]][1]);
                int ch_right = Math.abs(curr_right[0]-ph_nums[numbers[i]][0])+Math.abs(curr_right[1]-ph_nums[numbers[i]][1]);
                
                if(ch_left == ch_right && hand.equals("right")){
                    curr_right = ph_nums[numbers[i]];
                    answer += "R";
                }else if(ch_left == ch_right && hand.equals("left")){
                    curr_left = ph_nums[numbers[i]];
                    answer += "L";
                }else if(ch_left > ch_right){
                    curr_right = ph_nums[numbers[i]];
                    answer += "R";
                }else if(ch_left < ch_right){
                    answer += "L";
                    curr_left = ph_nums[numbers[i]];
                }
            }
        }
        return answer;
    }
}

File: Programmers/Level1/키패드 누르기/키패드 누르기.py
def solution(numbers, hand):
    answer = ''
    cordinate = [(3,1),(0,0),(0,1),(0,2),(1,0),(1,1),(1,2),(2,0),(2,1),(2,2)]
    L_current = (3,0)
    R_current = (3,2)
    
    for i in range(len(numbers)):
        if numbers[i] == 1 or numbers[i] == 4 or numbers[i] == 7:
            answer += 'L'
            L_current = cordinate[numbers[i]]
        elif numbers[i] == 3 or numbers[i] == 6 or numbers[i] == 9:
            answer += 'R'
            R_current = cordinate[numbers[i]]
        else:
            Cy, Cx = cordinate[numbers[i]]
            Ly, Lx = L_current 
            L_dis = abs(Ly-Cy)+abs(Lx-Cx)
            Ry, Rx = R_current
            R_dis = abs(Ry-Cy)+abs(Rx-Cx)
            
            if L_dis == R_dis:
                if hand == 'right':
                    R_current = cordinate[numbers[i]]
                    answer += 'R'
                else:
                    L_current = cordinate[numbers[i]]
                    answer += 'L'
            elif L_dis > R_dis:
                R_current = cordinate[numbers[i]]
                answer += 'R'
            else:
                L_current = cordinate[numbers[i]]
                answer += 'L'
            
    return answer
